### PR TITLE
Add scripts and configuration to upgrade L1 system config

### PR DESCRIPTION
**Pull request by henridevieux**
Adds scripts for the following:
1. Deploy new L1 systemconfig that has the following 2 changes:
    a. The max_gas_limit is `400_000_000`, instead of current value of `200_000_000`
    b. The version is set to `2.2.0+max-gas-limit-400M"`
2. Upgrade the existing systemconfig to the new implementation contract

The SystemConfig changes for implementation contract can be seen here: https://github.com/ethereum-optimism/optimism/pull/12879/files

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: mainnet/2024-11-18-increase-max-gas-limit/Makefile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	git reset --hard FETCH_HEAD
+
+
+.PHONY: deploy
+deploy:
+	$(GOPATH)/bin/eip712sign --ledger --hd-paths "m/44'/60'/$(LEDGER_ACCOUNT)'/0/0" -- \
+	forge script --rpc-url $(L1_RPC_URL) DeploySystemConfig --sig "sign(address)"
+
+.PHONY: sign-upgrade-cb
+sign-upgrade-cb:
```
Comment (henridevieux) [inline mainnet/2024-11-18-increase-max-gas-limit/Makefile:27]: Needs updating for all signers

File: mainnet/2024-11-18-increase-max-gas-limit/script/DeploySystemConfig.s.sol [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import "@eth-optimism-bedrock/contracts/L1/SystemConfig.sol";
+import "forge-std/Script.sol";
+
+contract DeploySystemConfig is Script {
+    function run() public {
+        vm.startBroadcast();
+        SystemConfig systemConfigImpl = new SystemConfig();
+	console.log("SystemConfig implementation deployed at: ", address(systemConfigImpl));
+	vm.stopBroadcast();
```
Comment (xenoliss) [inline mainnet/2024-11-18-increase-max-gas-limit/script/DeploySystemConfig.s.sol:12]:
```suggestion
        vm.startBroadcast();
        SystemConfig systemConfigImpl = new SystemConfig();
	    console.log("SystemConfig implementation deployed at: ", address(systemConfigImpl));
	    vm.stopBroadcast();
```

File: mainnet/2024-11-18-increase-max-gas-limit/.env [hunk 3]
```diff
@@ -0,0 +1,9 @@
+BASE_CONTRACTS_COMMIT=ed36aac52a19bdad6dee09c59e7241fe3a194160
+OPTIMISM_REPO=https://github.com/ethereum-optimism/optimism.git
+OPTIMISM_PR_ID=pull/12879/head
+
+SAFE_ADDRESS=0x7bb41c3008b3f03fe483b28b8db90e19cf07595c
+PROXY_ADMIN_ADDRESS=0x0475cBCAebd9CE8AfA5025828d5b98DFb67E059E
+SYSTEM_CONFIG_ADDRESS=0x73a79Fab69143498Ed3712e519A88a918e1f4072
+
+NEW_IMPLEMENTATION=0x3c768A33C473F664678b58C2253DB096b41F7cFC
```
Comment (0x00101010) [inline mainnet/2024-11-18-increase-max-gas-limit/.env:9]: https://etherscan.io/address/0x3c768a33c473f664678b58c2253db096b41f7cfc#code

**Review by xenoliss — APPROVED** [review]

File: mainnet/2024-11-18-increase-max-gas-limit/foundry.toml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+[profile.default]
+src = 'src'
+out = 'out'
+libs = ['lib']
+broadcast = 'records'
+fs_permissions = [{ access = "read-write", path = "./" }]
+optimizer = true
+optimizer_runs = 999999
+solc_version = "0.8.15"
+via-ir = false
```
Comment (mdehoog) [inline mainnet/2024-11-18-increase-max-gas-limit/foundry.toml:10]: maybe not in this PR, but we should set this as the default in the template

Comment (0x00101010) [reply mainnet/2024-11-18-increase-max-gas-limit/foundry.toml:10]: https://github.com/base-org/contract-deployments/pull/228

**Review by mdehoog — APPROVED** [review]

**Pull request merged**